.github/workflows/main.yml

```diff
@@ -718,7 +718,7 @@ jobs:
     timeout-minutes: 60
     env:
       BOOST_ROOT: ${{github.workspace}}/3rdparty/boost
-      BOOST_URL: https://boostorg.jfrog.io/artifactory/main/release/1.78.0/source/boost_1_78_0.7z
+      BOOST_URL: https://github.com/boostorg/boost/releases/download/boost-1.87.0/boost-1.87.0-b2-nodocs.7z
```

Comment:
There's an action that will install pre-compiled versions of boost. I know we don't use the compiled libraries, but using the action might make caching easier since they'll handle all of that, and simplify our workflow file here. I used the action in the PyPI repo if you want to see it, I'm on my phone so I don't have a link handy 😀

Reply:
Since this is the fix for some of the current CI breakages, I’d be in favor of pulling it out into its own PR.

Reply:
Here's where we're using it in the PyPI workflow: https://github.com/Cantera/pypi-packages/blob/a561cfa843dfe10b7682686447ef25cdb976dbc6/.github/workflows/python-package.yml#L210-L224 and here's the extension: https://github.com/MarkusJx/install-boost

Reply:
Using the current approach, restoring just the headers from the cache takes ~7 seconds. Restoring the full complement of compiled libraries from the cache takes ~36 seconds. I think the speed difference is worth sticking to the current approach.